### PR TITLE
feat: configurable products open and change

### PR DESCRIPTION
**Pull request by ivan-kalachikov**
### Jira-link:
https://virtocommerce.atlassian.net/browse/VCST-2093
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-2.11.0-pr-1478-b905-b9052847.zip

### CODE REVIEW

File: client-app/core/api/graphql/cart/mutations/changeCartConfiguredItem/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import type { ConfigurationSectionInput } from "@/core/api/graphql/types";
+
+type MutationVariablesType = {
+  lineItemId: string;
+  configurationSections: ConfigurationSectionInput[];
+  quantity: number;
+};
+
+export function useChangeCartConfiguredItemMutation() {
+  const { cultureName, currencyCode, storeId, userId } = toValue(useAllGlobalVariables());
```
Comment (NaMax66) [inline client-app/core/api/graphql/cart/mutations/changeCartConfiguredItem/index.ts:13]: import { globals } from "@/core/globals";

useAllGlobalVariables lets mark deprecated

File: client-app/shared/cart/components/add-to-cart.vue [hunk 2]
```diff
@@ -106,10 +106,12 @@ async function onChange() {
   let updatedCart: ShortCartFragment | undefined;
 
   const isAlreadyExistsInTheCart = !!lineItem;
-  const mode = isAlreadyExistsInTheCart && !isConfigurable.value ? AddToCartModeType.Update : AddToCartModeType.Add;
+  const mode = isAlreadyExistsInTheCart ? AddToCartModeType.Update : AddToCartModeType.Add;
 
   if (mode === AddToCartModeType.Update) {
-    updatedCart = await changeItemQuantity(lineItem!.id, enteredQuantity.value || 0);
+    updatedCart = isConfigurable.value
+      ? await changeCartConfiguredItem(lineItem!.id, enteredQuantity.value || 0, selectedConfigurationInput.value)
```
Comment (NaMax66) [inline client-app/shared/cart/components/add-to-cart.vue:113]: ! try not to use this

File: client-app/shared/cart/components/add-to-cart.vue [hunk 3]
```diff
@@ -129,7 +131,7 @@ async function onChange() {
 
   lineItem = clone(getLineItem(updatedCart?.items));
 
-  if (!lineItem) {
+  if (!lineItem && !isConfigurable.value) {
```
Comment (NaMax66) [inline client-app/shared/cart/components/add-to-cart.vue:134]: what if !lineItem && isConfigurable.value

File: client-app/shared/cart/components/add-to-cart.vue (hunk 2 above)
Comment (ivan-kalachikov) [reply client-app/shared/cart/components/add-to-cart.vue:113]: image
there is such a schema, which implicitly checked if lineItem exists, but we need to use ! as it too complicated for TypeScript to realize that :)

File: client-app/shared/cart/components/add-to-cart.vue [hunk 4]
```diff
@@ -63,10 +56,13 @@ interface IProps {
 }
 
 const product = toRef(props, "product");
-const { cart, addToCart, changeItemQuantity } = useShortCart();
+const { cart, addToCart, changeItemQuantity, changeCartConfiguredItem } = useShortCart();
 const { t } = useI18n();
 const ga = useGoogleAnalytics();
 const { translate } = useErrorsTranslator<ValidationErrorType>("validation_error");
+const { lineItemId: configurableLineItemId } = useUrlSearchParams<{ lineItemId: string | undefined }>("history", {
```
Comment (NaMax66) [inline client-app/shared/cart/components/add-to-cart.vue:63]: why useUrlSearchParams it is better then 
image

File: client-app/shared/cart/components/add-to-cart.vue [hunk 5]
```diff
@@ -101,66 +97,84 @@ function onInput(value: number): void {
...
-      eventType: "addToCart",
-      sessionId: cart.value?.id,
-      productId: product.value.id,
-      storeId: globals.storeId,
-    });
+async function updateOrAddToCart(lineItem: ShortLineItemFragment | undefined, mode: AddToCartModeType) {
+  if (mode === AddToCartModeType.Update && !!lineItem) {
+    return isConfigurable.value
+      ? await changeCartConfiguredItem(lineItem.id, enteredQuantity.value || 0, selectedConfigurationInput.value)
+      : await changeItemQuantity(lineItem.id, enteredQuantity.value || 0);
```
Comment (NaMax66) [inline client-app/shared/cart/components/add-to-cart.vue:121]: you sure needed call changeItemQuantity if enteredQuantity.value not positive?

File: client-app/shared/cart/components/add-to-cart.vue (hunk 4 above)
Comment (ivan-kalachikov) [reply client-app/shared/cart/components/add-to-cart.vue:63]: You are right, using useUrlSearchParams is a bit redundant here. Created utils function and used it instead.

File: client-app/shared/cart/components/add-to-cart.vue (hunk 5 above)
Comment (ivan-kalachikov) [reply client-app/shared/cart/components/add-to-cart.vue]: No needed indeed, just kept the same logic as it was. Now changed - if there is no enteredQuantity do nothing and return current short cart

**Review by NaMax66 — APPROVED** [review]

**Review by NaMax66 — APPROVED** [review]

**Pull request merged**